pkg/java/jar.go

```diff
@@ -69,6 +69,22 @@ func HashClass(jarFile string, className string) (ClassHash, error) {
 	}, nil
 }
 
+func ReadMethodByteCode(jarFile string, className string) ([][]byte, error) {
+	r, err := zip.OpenReader(jarFile)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
At this point, we know we have a `*zip.ReadCloser` that needs closing but we have two possible return paths in the function, so we need to use a defer and close.

The way to do it is to use named return values, which would look like `(codes [][]byte, err error)` and then swap out the error if there is one in the defer statement.

I whipped up a trivial abstract example: https://go.dev/play/p/L0_NMt_aAZR
An example in this codebase can also be found here: https://github.com/palantir/log4j-sniffer/blob/e9addc2bf6a3ba943b272017a0b99effa46761b9/pkg/archive/paths.go#L63